### ISSUE TITLE
element 0 of tensors does not require grad and does not have a grad_fn

### ISSUE THREAD

**Issue by wypdk** [superhighlevel/ChromaGan_Pytorch_Remake#8]
![image](https://user-images.githubusercontent.com/77193086/171330354-9c98b64a-0d16-4e54-bec5-cb444594fa28.png)
想知道这个问题怎么改

**Comment by superhighlevel** [comment]
Can you provide extra information about your dataset?

**Comment by wypdk** [comment]
for example？

**Comment by superhighlevel** [comment]
I'm not sure what is this problem is but i thinks because of dataset. Sorry for the late reply, if u solve this problem, don't you mind to share the solution>?

**Comment by ArsenalCheng** [comment]
It's because predAB is calculated under "with torch.no_grad", so predAB.requires_grad is False, so does trainAB, that is the problem. I think the author want to ignore the gradients of parameters of vgg, so set vgg.requires_grad to False and delete with torch.no_grad will solve this problem

**Comment by superhighlevel** [comment]
Updated code, please pull and run again
